catkin_virtualenv/cmake/build_venv.py

```diff
@@ -60,8 +62,8 @@ def delete_bytecode(directory):
         requirements_filename=args.requirements,
         upgrade_pip=True,
         pip_version="9.0.3",
-        use_system_packages=True,
-        python=find_executable('python3') if args.python3 else find_executable('python2'),
+        use_system_packages=args.use_system_packages,
+        python=find_executable('python' + args.python_version),
```

Comment:
If I'm reading the docs right, `find_executable` will return `None` if you ask for a non-existent version of python (say, someone types it wrong or messes up dependencies for a future development version). While the default value for the `python` kwarg here is None, it doesn't look like Deployment handles that case at all -- it'll blindly attempt to call subprocess on `[None, "-m", "venv"] `, which I doubt will return a clean error. It's probably worth checking whether this returns `None` and warning the user appropriately.

Reply:
fixed, good call